locale-utils.js

```diff
@@ -0,0 +1,69 @@
+"use strict";
+
+const fs = require("fs");
+const path = require("path");
+
+const { FluentBundle } = require("fluent");
+
+const mozlog = require("./log");
+
+
+const log = mozlog("locale-utils");
+
+const localesDir = path.join("public", "locales");
+const availableLanguages = [];
+const fluentBundles = {};
+
+
+class FluentError extends Error {
+  constructor(fluentID = null, ...params) {
+    console.log("FluentError constructor");
+    console.log("fluentID: ", fluentID);
+    console.log("params: ", ...params);
+    super(...params);
+
+    if (Error.captureStackTrace) {
+      console.log("Error.captureStackTrace");
+      Error.captureStackTrace(this, FluentError);
+    }
+
+    this.fluentID = fluentID;
+  }
+}
+
+
+const LocaleUtils = {
+  init() {
+    const languageDirectories = fs.readdirSync( localesDir ).filter(item => {
+      return (item[0] !== "." && item.length === 2)
```

Comment:
What does `item.length === 2` do?

It seems like it's checking if the file/folder name is 2 characters long.